internals/daemon/daemon_test.go

```diff
@@ -110,7 +110,9 @@ func (s *daemonSuite) TestAddCommand(c *C) {
 		GuestOK: true,
 		GET:     getCallback,
 	}
+
 	API = append(API, &command)
+	defer func() { API = API[:len(API)-1] }()
```

Comment:
Discussed with Paul. Would personally have favoured a more generic setup (backup) and tear-down (restore) approach, but given that modifying the API in tests at this point is unlikely going to common in the test suite, I am OK with this.

Reply:
Can we please do a slightly more careful job here and pop out &command, specifically?

Reply:
I've changed this to use Fred's solution as it doesn't require an additional search + delete function